go/controller/api/v1alpha1/autogenmemory_types.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2025.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	MemoryConditionTypeAccepted = "Accepted"
+)
+
+// MemoryProvider represents the memory provider type
+// +kubebuilder:validation:Enum=Pinecone
+type MemoryProvider string
+
+const (
+	Pinecone MemoryProvider = "Pinecone"
+)
+
+// PineconeConfig contains Pinecone-specific configuration options
+type PineconeConfig struct {
+	// The index host to connect to
+	// +required
```

Comment:
Should we support a distance threshold to control how similar the results need to be?